### ISSUE TITLE
Reactive Form and Observable Source

### ISSUE THREAD

**Issue by jtandrew** [ng2-ui/auto-complete#176]
Is it possible to use a Reactive Form solution and also connect to an Observable Source as well?

Here is my use case.
- Connect to a web service to get a list of 20000+ items
- User starts typing in the values and it narrows to the item they want
- Upon choosing from the first list they will get a second box to input a value that matches with the first item.  The second list of options ranges from 1 to 10 items.

Any help is much appreciated!!!

**Comment by allenhwkim** [comment]
You can use Observable  source for that case. Please create a plunker example to explain your issue.

**Comment by jtandrew** [comment]
@allenhwkim Thanks for the reply.  I will see if I can get the use case together in a plunker.  I have it working partially.  I am going to try and get it working properly with a long string first and will go from there.

**Comment by jtandrew** [comment]
@allenhwkim I used the source from model4 on https://rawgit.com/ng2-ui/auto-complete/master/app/index.html#/directive-test

Here is my code on my side:

`<input  ngui-auto-complete
        id="model4"
        [source]="siteSearch"
        no-match-found-text="No Match Found"
        list-formatter="formatted_address"
        path-to-data="results"
        loading-text="Getting Site List..."
        max-num-list="20"
        min-chars="2" />
      <br/>selected model4: {{model4 | json}}<br/><br/>`

`siteSearch = "http://tmppwport0006:3000/sites_only?site_cd=ilike.*:my_own_keyword";`

Here is the value of one of the calls:
http://tmppwport0006:3000/sites_only?site_cd=ilike.*1AT1

If I open that call in my browser or in Postman it works as I would expect but it is not working in the app.  See the message below.

Here is the message I getting returned:
Uncaught TypeError: Cannot read property 'slice' of undefined
    at SafeSubscriber.autoComplete.getRemoteData.subscribe._this.isLoading [as _next] (auto-complete.component.js:139)
    at SafeSubscriber.__tryOrUnsub (Subscriber.js:234)
    at SafeSubscriber.next (Subscriber.js:183)
    at Subscriber._next (Subscriber.js:125)
    at Subscriber.next (Subscriber.js:89)
    at MapSubscriber._next (map.js:83)
    at MapSubscriber.Subscriber.next (Subscriber.js:89)
    at MapSubscriber._next (map.js:83)
    at MapSubscriber.Subscriber.next (Subscriber.js:89)
    at XMLHttpRequest.onLoad (xhr_backend.js:77)
    at ZoneDelegate.invokeTask (zone.js:363)
    at Object.onInvokeTask (ng_zone.js:264)
    at ZoneDelegate.invokeTask (zone.js:362)
    at Zone.runTask (zone.js:166)
    at XMLHttpRequest.ZoneTask.invoke (zone.js:416)

**Comment by allenhwkim** [comment]
As I said, please use Observable source instead of url.

**Comment by jtandrew** [comment]
This is my getSiteList subscription. It calls my service and gets my list.

`this.sitesService.getSiteList()
          .subscribe(
            data => {const sites = [];
            for (let key in data) {
              sites.push(data[key]);
            }
            console.log(sites);
          }
        );
`
How would I get this into the source?  Sorry for the questions

**Comment by azeinda** [comment]
Got the same problem. It's damn hard to get it working. 
I've got an autocomplete that should display streets and narrow the results as the user starts typing.
The address service returns Address objects that contains the street name and geograph. coordinates.
When a street is selected in the autocomplete, i've got to set the coordinates input fields.

in template.html : 
<td>Rue : <input auto-complete
	formControlName="street"
	[(ngModel)]="camera.street" 
	[source]="suggestedAddresses"
	[list-formatter]="autocompleListFormatter"
	 value-property-name="id"
	display-property-name="street"
	class="form-control"
	 (valueChanged)="newAddressSelected($event)">
</td>

in component.ts : 

suggestedAddresses: Observable<Array<Address>>;

constructor() {
...

this.suggestedAddresses = this.cameraForm.controls['street'].valueChanges
        .debounceTime(400)
        .distinctUntilChanged()
        .switchMap(term => this.addressService.search(term));
}

Basically, the idea was to use your autocomplete component in collaboration with the observable example from here : https://blog.thoughtram.io/angular/2016/01/06/taking-advantage-of-observables-in-angular2.html

**Comment by allenhwkim** [comment]
Observable looks  not easy, but it's worth to master it.

Please take a look at this example to follow.
http://plnkr.co/edit/ExzNSh?p=preview

**Comment by jemerson3** [comment]
ERROR TypeError: _co.companies$.bind is not a function
